File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy = ListNode
        dummy.next = head
        temp = dummy
        while head:
            if head.val == val:
                temp.next = head.next
                head = head.next
            else:
                temp = head
                head = head.next
        return dummy.next
                
        

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode* dummy = new ListNode();
        ListNode* ans = dummy;
        while (head){
            if(head->val == val){
                dummy->next = head->next;
            }
            else{
                dummy->next = head;
                dummy = head;
            }
            head = head->next;
        }
        return ans->next;
    }
};